### ISSUE TITLE
[Training] The gradient builder has not been registered: Atan

### ISSUE THREAD

**Issue by cocotdf** [microsoft/onnxruntime#22407]
### Describe the issue

I realized that a certain number of nodes do not have gradients, specifically in this case, the Atan node. Is it planned to add support for it in future updates?

In the future, where should we send messages to indicate which nodes we would like gradients to be added for? Is there a specific procedure or channel to follow in order to submit such requests to the library developers?

### To reproduce

Try generating training artifacts for any ONNX model that includes the Atan node, similar to the one I provided.
[basic_model_with_atan.zip](https://github.com/user-attachments/files/17345154/basic_model_with_atan.zip)

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.18.1

### PyTorch Version

2.2.0

### Execution Provider

Default CPU
